components/SpotForm.js

```diff
@@ -0,0 +1,172 @@
+import { useState } from "react";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import { formTaskOptions } from "../helpers/formTaskOptions";
+import { formEquipmentOptions } from "../helpers/formEquipmentOptions";
+import { v4 as uuidv4 } from "uuid";
+
+export default function SpotForm({ addSpots }) {
+  const [formData, setFormData] = useState({
+    id: Date.now(),
+    Spotname: "",
+    Address: "",
+    Tasks: [],
+    Equipment: [],
+  });
+
+  const router = useRouter();
+
+  const handleOptionToggle = (option, field) => {
+    setFormData((prevFormData) => {
+      const selectedOptions = prevFormData[field];
+
+      if (selectedOptions.includes(option)) {
+        return {
+          ...prevFormData,
+          [field]: selectedOptions.filter(
+            (selectedOption) => selectedOption !== option
+          ),
+        };
+      } else {
+        return {
+          ...prevFormData,
+          [field]: [...selectedOptions, option],
+        };
+      }
+    });
+  };
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [name]: value,
+    }));
+  };
+
+  const onSubmitHandler = (event) => {
+    event.preventDefault();
+    const newSpot = {
+      id: formData.id,
+      Spotname: formData.Spotname,
+      Address: formData.Address,
+      Tasks: formData.Tasks,
+      Equipment: formData.Equipment,
+    };
+    const existingSpots = JSON.parse(localStorage.getItem("spots")) || [];
+    const updatedSpots = [...existingSpots, newSpot];
+    localStorage.setItem("spots", JSON.stringify(updatedSpots));
+    addSpots(newSpot);
+    setFormData({
+      id: Date.now(),
+      Spotname: "",
+      Address: "",
+      Tasks: [],
+      Equipment: [],
+    });
+    console.log(formData);
+    router.push("/spots");
+  };
+
+  return (
+    <FormSection>
+      <Form onSubmit={onSubmitHandler}>
+        <FormGroup>
+          <FormInputLabel htmlFor="SpotName">Spot Name:</FormInputLabel>
+          <FormInput
+            type="text"
+            name="Spotname"
+            value={formData.Spotname}
+            onChange={handleChange}
+          />
+        </FormGroup>
+        <FormGroup>
+          <FormInputLabel htmlFor="Address">Address:</FormInputLabel>
+          <FormInput
+            type="text"
+            name="Address"
+            value={formData.Address}
+            onChange={handleChange}
+          />
+        </FormGroup>
+        <fieldset key={uuidv4()}>
+          <legend>Tasks:</legend>
+          {formTaskOptions.map((task) => (
+            <FormLabel key={task.id}>
```

Comment:
Just use the task itself for the key if you don´t have an id.
The same should be applied to the equipments (line 109).